src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.cs

```diff
@@ -622,7 +628,7 @@ public unsafe bool IsPrefix(ReadOnlySpan<char> source, ReadOnlySpan<char> prefix
             else
             {
                 // Linguistic comparison requested and we don't need to special-case any args.
-#if TARGET_BROWSER
+#if TARGET_BROWSER || TARGET_OSX || TARGET_MACCATALYST || TARGET_IOS || TARGET_TVOS
```

Comment:
The code to set `matchLengthPtr` in `CompareInfo.ICU.cs` and `CompareInfo.OSX.cs` suggests that OSX does support match length.

Reply:
`matchLengthPtr` is sen on `OSX` only for `IndexOf`, `LasIndexOf` functions. For `IsPrefix` and `IsSuffix` functions for now it will not be supported on `OSX`.

Reply:
To get the correct `matchLengthPtr` locale-aware search is needed which is not supported for all cases on `OSX`.
Users will be interested in this  `matchLengthPtr` mostly when in the source string and target string letters have different representations, which is not implementable on `OSX` (this cases are also in IndexOf function). 
So with my current research I would say it is not implementable and reason is same as on WASM Linguistic comparison is not supported.

Reply:
Opened https://github.com/dotnet/runtime/issues/87767 for future investigations.
Added it to tracking issue for hybrid globalization work https://github.com/dotnet/runtime/issues/80689

Reply:
As @ilonatommy mentioned in below conversation
When disabling `matchLengthPtr` for WASM they did a research on frequency of usage of this parameter on GitHub out of our runtime libs and `matchLengthPtr` is used very rarely.

|     **API**     |                                                                                                    **All usages**                                                                                                   |                                                                                                                                          **Usages with matchLength**                                                                                                                                         |
|:---------------:|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------:|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------:|
|   **IndexOf**   |   [610 000](https://github.com/search?q=language%3AC%23++%2F%5C.IndexOf%2F++AND+NOT+%28path%3ACompareInfo.cs+OR+path%3AString.Manipulation.cs+OR+path%3A%2FSystem.Globalization%5C%2Ftests%5C%2F%2F%29&type=code)   |   [11](https://github.com/search?q=language%3AC%23++%2F%5C.IndexOf%5C%28%5B%5E%22%28%29%5D*%2C%5B%5E%22%28%29%5D*%2C%5B%5E%22%28%29%5D*%2C%5B%5E%22%28%29%5D*+out+%2F+AND+NOT+%28path%3ACompareInfo.cs+OR+path%3AString.Manipulation.cs+OR+path%3A%2FSystem.Globalization%5C%2Ftests%5C%2F%2F%29&type=code)  |
| **LastIndexOf** | [171 000](https://github.com/search?q=language%3AC%23++%2F%5C.LastIndexOf%2F++AND+NOT+%28path%3ACompareInfo.cs+OR+path%3AString.Manipulation.cs+OR+path%3A%2FSystem.Globalization%5C%2Ftests%5C%2F%2F%29&type=code) | [6](https://github.com/search?q=language%3AC%23++%2F%5C.LastIndexOf%5C%28%5B%5E%22%28%29%5D*%2C%5B%5E%22%28%29%5D*%2C%5B%5E%22%28%29%5D*%2C%5B%5E%22%28%29%5D*+out+%2F+AND+NOT+%28path%3ACompareInfo.cs+OR+path%3AString.Manipulation.cs+OR+path%3A%2FSystem.Globalization%5C%2Ftests%5C%2F%2F%29&type=code) |
|   **IsPrefix**  |    [2 300](https://github.com/search?q=language%3AC%23++%2F%5C.IsPrefix%2F++AND+NOT+%28path%3ACompareInfo.cs+OR+path%3AString.Manipulation.cs+OR+path%3A%2FSystem.Globalization%5C%2Ftests%5C%2F%2F%29&type=code)   |   [2](https://github.com/search?q=language%3AC%23++%2F%5C.IsPrefix%5C%28%5B%5E%22%28%29%5D*%2C%5B%5E%22%28%29%5D*%2C%5B%5E%22%28%29%5D*%2C%5B%5E%22%28%29%5D*+out+%2F+AND+NOT+%28path%3ACompareInfo.cs+OR+path%3AString.Manipulation.cs+OR+path%3A%2FSystem.Globalization%5C%2Ftests%5C%2F%2F%29&type=code)  |
|   **IsSuffix**  |     [732](https://github.com/search?q=language%3AC%23++%2F%5C.IsSuffix%2F++AND+NOT+%28path%3ACompareInfo.cs+OR+path%3AString.Manipulation.cs+OR+path%3A%2FSystem.Globalization%5C%2Ftests%5C%2F%2F%29&type=code)    |   [0](https://github.com/search?q=language%3AC%23++%2F%5C.IsSuffix%5C%28%5B%5E%22%28%29%5D*%2C%5B%5E%22%28%29%5D*%2C%5B%5E%22%28%29%5D*%2C%5B%5E%22%28%29%5D*+out+%2F+AND+NOT+%28path%3ACompareInfo.cs+OR+path%3AString.Manipulation.cs+OR+path%3A%2FSystem.Globalization%5C%2Ftests%5C%2F%2F%29&type=code)  |

Reply:
Why is locale-aware search needed to get the exact length of the substring in the `sourceString` that is matched by `searchString`?
The readme mentioned that exactly length cannot be calculated because of `As we are normalizing strings to be able to cut them, we cannot calculate the match length on the original strings`.
Is it because we normalize the source string that we cannot calculate the length because the expected match length is based on the number of unicode code points?

I'm wondering if it is at all possible to
```
// If no match with same normalization form, early return
if (not source.precomposed match search.precomposed)
    return 0;

// Iterate over each character and add number of unicode code points
foreach (search.char)
   if (source[i] match search.char.precomposed)
       matchLength += search.char.precomposed number of code points
   else if (source[i] match search.char.decomposed)
       matchLength += search.char.decomposed number of code points
```

Reply:
Yes it is not supported because we normalize the strings before checking `IsPrefix` or `IsSuffix`.

Implementing through iteration will create same issues (not covered cases) that we have in `IndexOf` function and will be way more slower.
One more solution will be to call `IndexOf` function inside `IsPrefix` for example. 
But note that doing this we will lose full coverage of cases for `IsPrefix` and `IsSuffix` functions and have more complex implementation. For further investigation I have created ticket already https://github.com/dotnet/runtime/issues/87767
I prefer to keep current implementation and not support `matchLengthPtr` for now as it is not so used parameter.